### ISSUE TITLE
Segmentation fault in geosop largestEmptyCircleBdy

### ISSUE THREAD

**Issue by gabe-sherman** [libgeos/geos#1017]
I am an undergraduate student exploring automatic fuzzing harness generation for open source api's. We found that executing geosop with these arguments:

`
 ./geosop -a "POLYGON ((0 0, 4 0,00 60, 1 4, 0 4, 444446444444 4,10 4, 0 4, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))" -b "POLYGON ((0 0, 4 0,00 60, 1 4, 0 4, 444446444444 4,10 4, 0 4, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))" largestEmptyCircleBdy 2
`

results in a segmentation fault.

Here's the asan report of the fuzzing harness used to find this:

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==87601==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7ffff78d37da bp 0x000000000000 sp 0x7fffffffdd00 T0)
==87601==The signal is caused by a READ memory access.
==87601==Hint: address points to the zero page.
    #0 0x7ffff78d37da in std::__uniq_ptr_impl<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits>, std::default_delete<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits> > >::_M_ptr() const /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unique_ptr.h:191:51
    #1 0x7ffff78d37da in std::unique_ptr<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits>, std::default_delete<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits> > >::get() const /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unique_ptr.h:462:21
    #2 0x7ffff78d37da in std::unique_ptr<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits>, std::default_delete<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits> > >::operator->() const /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unique_ptr.h:455:9
    #3 0x7ffff78d37da in geos::operation::distance::IndexedFacetDistance::nearestLocations(geos::geom::Geometry const*) const /home/gabe/harness_test/aflgeos/geos/src/operation/distance/IndexedFacetDistance.cpp:87:20
    #4 0x7ffff78d3033 in geos::operation::distance::IndexedFacetDistance::nearestPoints(geos::geom::Geometry const*) const /home/gabe/harness_test/aflgeos/geos/src/operation/distance/IndexedFacetDistance.cpp:99:57
    #5 0x7ffff7732aac in geos::algorithm::construct::IndexedDistanceToPoint::nearestPoints(geos::geom::Point const&) /home/gabe/harness_test/aflgeos/geos/src/algorithm/construct/IndexedDistanceToPoint.cpp:65:27
    #6 0x7ffff773a2b4 in geos::algorithm::construct::LargestEmptyCircle::compute() /home/gabe/harness_test/aflgeos/geos/src/algorithm/construct/LargestEmptyCircle.cpp:276:47
    #7 0x7ffff7738923 in geos::algorithm::construct::LargestEmptyCircle::getRadiusLine() /home/gabe/harness_test/aflgeos/geos/src/algorithm/construct/LargestEmptyCircle.cpp:102:5
    #8 0x7ffff7f701c2 in GEOSLargestEmptyCircle_r::$_57::operator()() const /home/gabe/harness_test/aflgeos/geos/capi/geos_ts_c.cpp:1356:27
    #9 0x7ffff7f701c2 in _Z7executeIZ24GEOSLargestEmptyCircle_rE4$_57LDn0EEDTclfp0_EEP20GEOSContextHandle_HSOT_ /home/gabe/harness_test/aflgeos/geos/capi/geos_ts_c.cpp:427:16
    #10 0x7ffff7f701c2 in GEOSLargestEmptyCircle_r /home/gabe/harness_test/aflgeos/geos/capi/geos_ts_c.cpp:1354:16
    #11 0x5555556312a3 in main /home/gabe/harness_test/harness_test/APIS/geosfuzzing/fuzzingcampaigns/harness307-winter/test.c:49:48
    #12 0x7ffff7c29d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #13 0x7ffff7c29e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #14 0x555555573494 in _start (/home/gabe/harness_test/harness_test/APIS/geosfuzzing/fuzzingcampaigns/harness307-winter/test.out+0x1f494) (BuildId: bbf9f400e1fd5060ce49b51779b92b28178a47f9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unique_ptr.h:191:51 in std::__uniq_ptr_impl<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits>, std::default_delete<geos::index::strtree::TemplateSTRtree<geos::operation::distance::FacetSequence const*, geos::index::strtree::EnvelopeTraits> > >::_M_ptr() const
==87601==ABORTING
```

Test Environment

Ubuntu 22.04, 64bit
Harness written using geos capi
Geos Version Commit: commit 28d70a2e4582edcd3625659f70e31d1853423875

**Comment by dr-jts** [comment]
I can't reproduce this with the current code.  It's possible it was fixed by #1004

**Comment by gabe-sherman** [comment]
I believe you're right, thanks for checking it out!
